sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/LogsQueryAsyncClient.java

```diff
@@ -190,25 +239,22 @@ private Context updateContext(Duration serverTimeout, Context context) {
             LogsQueryResult logsQueryResult = getLogsQueryResult(queryResults.getTables(),
                     queryResults.getStatistics(), queryResults.getRender(), queryResults.getError());
             LogsBatchQueryResult logsBatchQueryResult = new LogsBatchQueryResult(singleQueryResponse.getId(),
-                    singleQueryResponse.getStatus(), logsQueryResult);
+                    singleQueryResponse.getStatus(), logsQueryResult.getAllTables(), logsQueryResult.getStatistics(),
+                    logsQueryResult.getVisualization(), logsQueryResult.getError());
             batchResults.add(logsBatchQueryResult);
         }
         batchResults.sort(Comparator.comparingInt(o -> Integer.parseInt(o.getId())));
-        return new SimpleResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), logsBatchQueryResultCollection);
+        return new SimpleResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), logsBatchQueryResults);
     }
 
-    private LogsQueryError mapLogsQueryError(ErrorInfo errors) {
+    private HttpResponseError mapLogsQueryError(ErrorInfo errors) {
         if (errors != null) {
-            List<LogsQueryErrorDetail> errorDetails = Collections.emptyList();
+            List<HttpResponseError> errorDetails = Collections.emptyList();
             if (errors.getDetails() != null) {
                 errorDetails = errors.getDetails()
                         .stream()
-                        .map(errorDetail -> new LogsQueryErrorDetail(errorDetail.getCode(),
-                                errorDetail.getMessage(),
-                                errorDetail.getTarget(),
-                                errorDetail.getValue(),
-                                errorDetail.getResources(),
-                                errorDetail.getAdditionalProperties()))
+                        .map(errorDetail -> new HttpResponseError(errorDetail.getCode(), errorDetail.getMessage())
```

Comment:
>errorDetail [](http://example.com/codeflow?start=29&length=11)

Why remove the strongly typed error here and make it generic? Do we not see value in having a new type?

Reply:
We have a common error shape defined by the REST API guidelines. All services are expected to use the same error structure. So, we want to define this type once in core and reuse it in all services instead of creating new types in each client library.